### PR TITLE
CP-28406: improve logging and error checking in clustering functions

### PR DESCRIPTION
**Pull request by edwintorok**
This was useful in debugging problems with @minishrink PR.

### CODE REVIEW

File: ocaml/xapi/xapi_cluster.ml [hunk 1]
```diff
@@ -85,21 +84,13 @@ let destroy ~__context ~self =
...
-    Xapi_stdext_monadic.Opt.iter (fun ch ->
-      Db.Cluster_host.destroy ~__context ~self:ch
-    ) cluster_host;
-    Db.Cluster.destroy ~__context ~self;
-    D.debug "Cluster destroyed successfully";
-    set_ha_cluster_stack ~__context;
-    Xapi_clustering.Daemon.disable ~__context
-  | Result.Error error ->
-    D.warn "Error occurred during Cluster.destroy";
-    handle_error error
```
Comment (minishrink) [inline ocaml/xapi/xapi_cluster.ml]: This is good, we were unnecessarily duplicating code.

File: ocaml/xapi/xapi_cluster_host.ml [hunk 2]
```diff
@@ -136,43 +136,42 @@ let create ~__context ~cluster ~host ~pif =
...
-  assert_cluster_host_has_no_attached_sr_which_requires_cluster_stack ~__context ~self;
-  let result = Cluster_client.LocalClient.destroy (rpc ~__context) dbg in
-  match result with
-  | Result.Ok () ->
-    Db.Cluster_host.destroy ~__context ~self;
-    debug "Cluster_host.force_destroy was successful";
-    Xapi_clustering.Daemon.disable ~__context
-  | Result.Error error ->
-    warn "Error occurred during Cluster_host.force_destroy";
-    handle_error error
```
Comment (minishrink) [inline ocaml/xapi/xapi_cluster_host.ml]: Also nicely done!

File: ocaml/xapi/xapi_clustering.ml [hunk 3]
```diff
@@ -200,6 +200,8 @@ let assert_cluster_host_has_no_attached_sr_which_requires_cluster_stack ~__conte
     ) srs
   then raise Api_errors.(Server_error (cluster_stack_in_use, [ cluster_stack ]))
 
+let daemon_enabled = ref false
```
Comment (minishrink) [inline ocaml/xapi/xapi_clustering.ml]: Do we want a helper to query this?

Comment (minishrink) [reply ocaml/xapi/xapi_clustering.ml]: Actually ignore that, the code is good to merge

**Comment by edwintorok** [comment]
Just realized that we do need to prevent Clusters without cluster hosts, otherwise pool autojoin won't work, and in fact a Cluster object without cluster hosts is unusable.

**Comment by edwintorok** [comment]
We should merge https://github.com/xapi-project/xen-api/pull/3607 first and then this one needs to be rebased.

**Comment by edwintorok** [comment]
Rebased on latest branch.

**Pull request merged**